apps/api/test/06-report-6.e2e-spec.ts

```diff
@@ -0,0 +1,176 @@
+import { INestApplication } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import request from 'supertest';
+
+import { IENApplicantCreateUpdateDTO, STATUS } from '@ien/common';
+import { AppModule } from 'src/app.module';
+import { AuthGuard } from 'src/auth/auth.guard';
+import { URLS } from './constants';
+import { canActivate } from './override-guard';
+import {
+  getApplicant,
+  getIndexOfStatus,
+  getJob,
+  getMilestone,
+  RECRUITMENT_STAGE_STATUSES,
+} from './report-util';
+import { IENHaPcn } from 'src/applicant/entity/ienhapcn.entity';
+
+describe('Report 6 - Registrants in Recruitment Stage', () => {
+  let app: INestApplication;
+  let jobTempId = '';
+  let applicantStatusId = 'NA';
+  let applicantId: string;
+  let HA: IENHaPcn[] = [];
+
+  let refCheckIndex = 0;
+  let intvwIndex = 0;
+  let jobAcceptedIndex = 0;
+
+  let lastHa = '';
+
+  const PHSA = 'Provincial Health Services';
+  const NHA = 'Northern Health';
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    })
+      .overrideGuard(AuthGuard)
+      .useValue({ canActivate })
+      .compile();
+
+    app = moduleFixture.createNestApplication();
+    await app.init();
+
+    const { body: authorities } = await request(app.getHttpServer()).get('/ienmaster/ha-pcn');
+    // remove 'Authority' from end of HA strings
+    authorities.forEach((e: IENHaPcn) => {
+      e.title = e.title.substring(0, e.title.lastIndexOf(' '));
+    });
+    HA = authorities;
+    lastHa = HA[HA.length - 1].title;
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  const getReport6 = async () => {
+    const { body } = await request(app.getHttpServer()).get(URLS.REPORT6);
+    return body;
+  };
+
+  const addApplicant = async (applicant: IENApplicantCreateUpdateDTO) => {
+    const { body } = await request(app.getHttpServer()).post('/ien').send(applicant);
+    return body;
+  };
+
```

Comment:
Can addJob/addMilestone/addApplicant be moved to the util file and used by other tests?

Reply:
yes, I moved them in my report 7 test branch, but I think a separate PR to replace all the requests to use these fn would be best